typeinfo-less.md

```diff
@@ -0,0 +1,362 @@
+
+Order arrays first internally by element type, then by rank, then by rank
+bounds, lowest first.
+
+So the order of the following, for a given type T:
+
+```cpp
+T[10]
+T[11]
+T[10][2]
+T[3][2]
+```
+
+would be ordered `T[10] < T[11] < T[3][2] < T[10][2]`
+
+Class types will be ordered according to the rules below, see [ordering-classes]
+
+### namespaces
+
+Namespaces will also be ordered alphabetically.
+
+Given:
+```cpp
+namespace foo {
+  struct i;
+}
+
+namespace bar {
+  struct i;
+}
+```
+
+The order of the two structs w/ type `i` types would be `bar::i < foo::i`
+
+### anonymous namespace
+The anonymous namespace will be ordered after the global namespace but before
+any user namespaces.
+
+### Non Templated User Defined Types
+
+Simple User Defined Types can be defines as either a `struct` or a `class` that
+has been declared without a template parameter. As far as ordering is concerned,
+structs and classes should be treated the same.
+
+User defined types will be greater than native types.
+
+Ordering these is simple, since we cannot redeclare two types with the same
+name, we'll just order simple user defined types alphabetically. 
+
+```cpp
+struct Apple {};
+class Banana {};
+struct Carrot {};
+```
+
+Would be ordered as `Apple < Banana < Carrot`
+
+# Value Ordering
+Values can be used to create template specializations. Take, for example:
+
+```cpp
+template <int N>
+class Foo;
+....
+Foo<1>;
+Foo<2>;
+```
+
+To account for this, we must define an order for value. Values will be ordered
+first by their types, then by their individual values.
+
+so in the case of 
+```cpp
+Foo<1>
+Foo<2>
+```
+
+`Foo<1> < Foo<2>`
+
+When ordering types with value template arguments, `typeinfo` will use the 
+`operator<=>`
+
+Values without `operator<=>` cannot be compared, and will cause a compilation 
+error. An example of this would be 
+
+```cpp
+struct Foo {
+  int i;
+  // Notice, no way to order two Foos
+  constexpr Foo(int i_) : i{i_} {};
+  
+  friend auto operator<(Foo const& lhs, Foo const& rhs) -> bool = delete;
+};
+
+template <Foo T>
+struct Bar {
+    int j = T.i;
+};
+```
+
+## Ordering Templated Classes
+
+Lets start with the simple, non-recursive case. Templated classes are ordered
+by;
+1. Class name, alphabetically
+2. Order of template arguments applied left to right.
+
+For example, given:
+```cpp
+template <typename T, typename U>
+struct Apple;
+
+struct Banana;
+struct Carrot;
+
+Apple<Banana, Carrot>;
+Apple<Banana, Banana>;
+Apple<Carrot, Carrot>;
+```
+
+would be ordered `Apple<Banana, Carrot> < Apple<Banana, Banana> < Apple<Carrot, Carrot>`.
+
+The same rules apply for recursive templates:
+
+For example, given:
+```cpp
+template <typename T>
+struct Foo;
+
+Foo<int>;
+Foo<Foo<int>>;
+```
+
+would be ordered `Foo<int> < Foo<Foo<int>>`, because `int` and `Foo<int>` 
+template arguments are compared, and `int < Foo<int>`.
+
+## Non-Templated User Defined Functions
```

Comment:
"the order of function types". You don't have to talk about user-defined here. All function types are the same.